config.yaml

```diff
@@ -134,3 +134,27 @@ options:
     type: boolean
     description: Configure Discourse to use https.
     default: false
+  reply_by_email_address:
+    type: string
+    description: Replies email in the form of replies+%{reply_key}@example.com.
+    default: ""
+  pop3_polling_username:
+    type: string
+    description: Username of the email account like replies@example.com.
+    default: ""
+  pop3_polling_password:
+    type: string
+    description: The password of that email account.
+    default: ""
+  pop3_polling_host:
+    type: string
+    description: Polling host such as pop.gmail.com.
+    default: ""
+  pop3_polling_enabled:
```

Comment:
What happens if this flag is enabled but the other configuration options are not provided? Shouldn't there be some sort of non-emptiness check? What does the GUI do in this case?

Reply:
Passing some environment variables has the same effect as using the GUI, so discourse will not enable the polling in that case but will not complain (we may see some alert in the admin settings)

Reply:
I agree that we should probably add some logic to put a warning in the status of the charm, or an error

Reply:
We might think about this, though like Niels said - Discourse should handle that. The best we could do is to put a warning maybe, as you could configure these settings in any order and once you have them, polling will work. If not, we are guiding the user to set them all at once, like `juju config a=x b=y..` on one line - which is also an option I guess, then we need to update the documentation to reflect that this is how we expect the charm to be configured in terms of polling.